messagebus/messagebus_test.go

```diff
@@ -304,6 +304,75 @@ var _ = Describe("Messagebus test Suite", func() {
 			})
 		})
 	})
+
+	Describe("SendMessage for h2 route", func() {
+		const (
+			topic             = "router.registrar"
+			host              = "some_host"
+			privateInstanceId = "some_id"
+		)
+
+		var (
+			route config.Route
+		)
+
+		BeforeEach(func() {
+			err := messageBus.Connect(messageBusServers, nil)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			port := 12345
+
+			route = config.Route{
+				Name:                "some_name",
+				Port:                &port,
+				TLSPort:             &port,
+				Protocol:            "http2",
+				URIs:                []string{"uri1", "uri2"},
+				RouteServiceUrl:     "https://rs.example.com",
+				Tags:                map[string]string{"tag1": "val1", "tag2": "val2"},
+				ServerCertDomainSAN: "cf.cert.internal",
+			}
+		})
+
+		It("send messages", func() {
+			registered := make(chan string)
+			testSpyClient.Subscribe(topic, func(msg *nats.Msg) {
+				registered <- string(msg.Data)
+			})
+
+			// Wait for the nats library to register our callback.
+			// We use a sleep because there's no way to know that the callback was
+			// registered successfully (e.g. they don't provide a channel)
+			time.Sleep(20 * time.Millisecond)
+
+			err := messageBus.SendMessage(topic, host, route, privateInstanceId)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Assert that we got the right message
+			var receivedMessage string
+			Eventually(registered, 2).Should(Receive(&receivedMessage))
+
+			expectedRegistryMessage := messagebus.Message{
+				URIs:                route.URIs,
+				Host:                host,
+				Port:                route.Port,
+				Protocol:            route.Protocol,
+				TLSPort:             route.TLSPort,
+				RouteServiceUrl:     route.RouteServiceUrl,
+				Tags:                route.Tags,
+				ServerCertDomainSAN: "cf.cert.internal",
+			}
+
+			var registryMessage messagebus.Message
+			err = json.Unmarshal([]byte(receivedMessage), &registryMessage)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Expect(registryMessage.URIs).To(Equal(expectedRegistryMessage.URIs))
+			Expect(registryMessage.Port).To(Equal(expectedRegistryMessage.Port))
+			Expect(registryMessage.RouteServiceUrl).To(Equal(expectedRegistryMessage.RouteServiceUrl))
+			Expect(registryMessage.Tags).To(Equal(expectedRegistryMessage.Tags))
```

Comment:
shouldn't we check for the protocol here?

Reply:
Done